src/ucp/wireup/wireup_cm.c

```diff
@@ -743,9 +741,7 @@ static void ucp_cm_client_connect_cb(uct_ep_h uct_cm_ep, void *arg,
 err_free_arg:
     ucs_free(progress_arg);
 err_out:
-    UCS_ASYNC_BLOCK(&worker->async);
```

Comment:
assert is locked?

Reply:
we're checking it in `UCP_EP_CM_CALLBACK_ENTER` macro at the begging of the function

Reply:
nice.